documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,331 @@
+
+### 1.4 Removing the order
+
+***Main scenario:***
+* The user, while in the list of orders, presses the "Delete" button in the selected order line;
+* If the order can be removed, a confirmation dialog is displayed; 
+* The user confirms the removal of the order; 
+* Record is deleted from database; 
+* If error occurs, then error message displays; 
+* If order record is successfully deleted, then list of orders without deleted records is displaying.
+ 
+
+![](delete_order.PNG)
+<br>Pic. 1.4 Delete order dialog.
+
+***Cancel operation scenario:****
+* User is in display mode of orders list and press “Delete” button; 
+* Application displays confirmation dialog “Please confirm delete order?”; 
+* User press “Cancel” button; 
+* List of orders without changes is displaying. 
+
+
+## 2. Clients
+
+### 2.1 Display list of Clients
+
+This mode is intended for viewing and editing the clients list.
+
+***Main scenario:***
+* User selects item “Clients”; 
+* Application displays list of clients.
+
+![](Clients.PNG)
+<br>Pic 2.1 View the clients list.
+
+The list displays the following columns:
+* Last name – client’s last name; 
+* First name – client’s first name; 
+* Phone - the client's phone number;
+* Registration date – client’s date registration; 
+* Passport number – the unique number of the client's passport;
+
+
+***Filtering by date:***
+* In the clients list view mode, the user sets a date filter and presses the refresh list button (to the right of the date entry field); 
+* The application will show the clients only for a certain period of time.
+
+### 2.2 Add client
+
+***Main scenario:***
+* User clicks the “Add” button in the clients list view mode; 
+* Application displays form to enter client data; 
+* User enters client’s data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying; 
+* If new client record is successfully added, then list of clients with added records is displaying.
+
+***Cancel operation scenario:***
+* User clicks the “Add” button in the clients list view mode; 
+* Application displays form to enter client’s data; 
+* User enters client’s data and presses “Cancel” button; 
+* Data don’t save in data base, then list of clients records is displaying to user. 
+* If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+
+![](add_client.PNG)
+<br>Pic. 2.2 Add client.
+
+When adding a client, the following details are entered:
+* Last name – client’s last name; 
+* First name – client’s first name; 
+* Phone - the client's phone number;
+* Registration date – client’s date registration; 
+* Passport number – the unique number of the client's passport;
+
+Constraints for data validation:
+* Last name – maximum length of 45 characters; 
```

Comment:
In previous view you have defined that max length is 90 - and here - 45. Just a note. Again. you should not validate these data in `Orders` view.